### PR TITLE
No chequebook

### PR DESCRIPTION
**Pull request by svetomir**

### CODE REVIEW

File: pkg/orchestration/k8s/nodegroup.go [hunk 1]
```diff
@@ -69,6 +70,24 @@ func (g *NodeGroup) AddNode(name string, o orchestration.NodeOptions) (err error
...
 	})
 
+	// pregenerate Swarm key if needed
+	if o.SwarmKey == "" && config.Transaction == "" && (!config.SwapEnable || config.ChequebookEnable) {
+		swarmKey, overlay, err := utils.CreateSwarmKey(config.Password)
+		if err != nil {
+			return fmt.Errorf("create Swarm key for node %s: %w", name, err)
+		}
+
+		txHash, err := g.cluster.swap.AttestOverlayEthAddress(ctx, overlay)
```
Comment (ralph-pichler) [inline pkg/orchestration/k8s/nodegroup.go:80]: make sure there is a log line for this result. either here or inside the geth service.

File: pkg/orchestration/k8s/nodegroup.go [hunk 2]
```diff
@@ -69,6 +70,24 @@ func (g *NodeGroup) AddNode(name string, o orchestration.NodeOptions) (err error
 		Restricted:          config.Restricted,
 	})
 
+	// pregenerate Swarm key if needed
+	if o.SwarmKey == "" && config.Transaction == "" && (!config.SwapEnable || config.ChequebookEnable) {
```
Comment (ralph-pichler) [inline pkg/orchestration/k8s/nodegroup.go:74]: the last clause is wrong. should be `(!config.SwapEnable || !config.ChequebookEnable)`

File: pkg/orchestration/k8s/nodegroup.go (hunk 1 above)
Comment (svetomir) [reply pkg/orchestration/k8s/nodegroup.go:80]: Done.

File: pkg/orchestration/k8s/nodegroup.go (hunk 2 above)
Comment (svetomir) [reply pkg/orchestration/k8s/nodegroup.go]: Done.

**Review by ralph-pichler — APPROVED** [review]

**Pull request merged**